test/prompt/test_prompt_template.py

```diff
@@ -43,6 +46,35 @@ def test_prompt_template_repr():
     assert str(p) == desired_repr
 
 
+@pytest.mark.unit
+def test_prompt_template_deserialization():
```

Comment:
Can we try and make this a minimal `Pipeline._load_or_get_component()` unit test?

Also this is not really testing anything from `PromptTemplate` or `PromptNode`, the fact that we didn't even edit any of them makes it pretty clear. We should move it in another file.

Reply:
I kept the `PromptTemplate`-specific test as it is quite special what we're doing there:
- use a nested component
- nested component references its own name as param
- nested component nests another component

All three should be tested in isolation already. But testing it alltogether should give us additional safety over the happy path of using `PromptTemplate` in pipelines.